### PR TITLE
Python: Model `rsa`

### PR DESCRIPTION
**Pull request by RasmusWL**

### CODE REVIEW

File: python/ql/src/semmle/python/frameworks/Rsa.qll [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    RsaSignCall() { this = API::moduleImport("rsa").getMember("sign").getACall() }
+
+    override Cryptography::CryptographicAlgorithm getAlgorithm() {
+      // signature part
+      result.getName() = "RSA"
+      or
+      // hashing part
+      exists(StrConst str, DataFlow::Node hashNameArg |
+        hashNameArg in [this.getArg(2), this.getArgByName("hash_method")] and
+        DataFlow::exprNode(str).(DataFlow::LocalSourceNode).flowsTo(hashNameArg) and
```
Comment (yoff) [inline python/ql/src/semmle/python/frameworks/Rsa.qll:74]:
```suggestion
        DataFlow::exprNode(str) = hashNameArg.getALocalSource() and
```
I slightly prefer this form, as subtly implied in [the documentation](https://codeql.github.com/docs/codeql-language-guides/analyzing-data-flow-in-python/#examples).

File: python/ql/src/semmle/python/frameworks/Rsa.qll [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+   * See https://stuvel.eu/python-rsa-doc/reference.html#rsa.compute_hash
+   */
+  class RsaComputeHashCall extends Cryptography::CryptographicOperation::Range,
+    DataFlow::CallCfgNode {
+    RsaComputeHashCall() { this = API::moduleImport("rsa").getMember("compute_hash").getACall() }
+
+    override Cryptography::CryptographicAlgorithm getAlgorithm() {
+      exists(StrConst str, DataFlow::Node hashNameArg |
+        hashNameArg in [this.getArg(1), this.getArgByName("method_name")] and
+        DataFlow::exprNode(str).(DataFlow::LocalSourceNode).flowsTo(hashNameArg) and
```
Comment (yoff) [inline python/ql/src/semmle/python/frameworks/Rsa.qll:117]:
```suggestion
        DataFlow::exprNode(str) = hashNameArg.getALocalSource() and
```

**Review by yoff — CHANGES_REQUESTED** [review]
Nice, shows the strength of previous modelling. Only minor nitpicks.

File: python/ql/src/semmle/python/frameworks/Rsa.qll (hunk 1 above)
Comment (RasmusWL) [reply python/ql/src/semmle/python/frameworks/Rsa.qll:74]: Yeah I think taus recently did these rewrites as well. I obviously just copied from the old code :smile:

**Review by yoff — APPROVED** [review]
LGTM

**Pull request merged**